EKF/EKFGSF_yaw.cpp

```diff
@@ -73,18 +68,16 @@ void EKFGSF_yaw::update(const Vector3f del_ang, // IMU delta angle rotation vect
 			}
 
 			if (!bad_update) {
+				float total_weight = 0.0f;
 				// calculate weighting for each model assuming a normal distribution
 				for (uint8_t model_index = 0; model_index < N_MODELS_EKFGSF; model_index ++) {
-					newWeight[model_index] = fmaxf(gaussianDensity(model_index) * _ekf_gsf[model_index].W, 0.0f);
-					total_w += newWeight[model_index];
+					_model_weights(model_index) = fmaxf(gaussianDensity(model_index) * _model_weights(model_index), 0.0f);
+					total_weight += _model_weights(model_index);
 				}
 
 				// normalise the weighting function
-				if (_ekf_gsf_vel_fuse_started && total_w > 1e-15f && !bad_update) {
-					float total_w_inv = 1.0f / total_w;
-					for (uint8_t model_index = 0; model_index < N_MODELS_EKFGSF; model_index ++) {
-						_ekf_gsf[model_index].W = newWeight[model_index] * total_w_inv;
-					}
+				if (total_weight > 1e-15f) {
+					_model_weights /= total_weight;
 				}
 
 				// Enforce a minimum weighting value. This was added during initial development but has not been needed
```

Comment:
@priseborough Is this minimum weight computation still necessary? Could we not just bound it [here](https://github.com/PX4/ecl/pull/769/files#diff-9a4cd2c2327ab18e053e90ec5eb0accaR74)

Reply:
After normalisation the minimum value would be changed. After we get through a period if wider use/testing I'm hoping to remove the min weight parameter and that block of code.